### ISSUE TITLE
protoc-micro: command not found

### ISSUE THREAD

**Issue by mid-kid** [microg/build#1]
I have no idea where that command exists, so I replaced it with "protoc".
Result:

```
protoc-gen-javamicro: program not found or is not executable
--javamicro_out: protoc-gen-javamicro: Plugin failed with status code 1.
```

I haven't really looked into it.

**Comment by mar-v-in** [comment]
Use https://code.google.com/p/micro-protobuf/

**Comment by mid-kid** [comment]
OK, since there's literally nowhere from where I can download a prebuilt package (Nor Debian, Nor AUR), I had to compile that myself. No biggie, llinked statically, had to change protoc-micro into protoc in the script and add it to PATH.
Since the script doesn't get the CLASSPATH variable, and has no switch for setting a custom, I had to manually change the CLASSPATH in the script for apache http. So far, so good.
Also had to add the $sdk_path and $sdk_version into ./build.sh since they don't get detected correctly.
GoogleServicesFrameworkInclude doesn't use android.os for some reason; had to fix that too.
`Note: /home/mid-kid/Stuff/programming/microg/build/sources/Phonesky/src/org/microg/playstore/StoreApplication.java uses or overrides a deprecated API.`
...and then, I found a file named `local-config.microg-build.sample` :expressionless:
Created a new signing key... All fine.

```
Warning: 
No -tsa or -tsacert is provided and this jar is not timestamped. Without a timestamp, users may not be able to validate this jar after the signer certificate's expiration date (2041-06-06) or after any future revocation date.
```

And to top it off, Phonesky doesn't even work (yet). YAY!

Instead of writing your own build system, maybe apache maven can help you. :wink: (Or maybe even ant has some dependency support, I haven't really looked)

**Comment by mar-v-in** [comment]
Sorry for not giving you a convenient build system. 

I originally created the build system for my personal use and it was about a 50 line bash script but grow as i needed more features. It was never intended to be used by others and i wanted to replace it with another build system long time ago. This is also the reason i never documented it.

This version of Phonesky currently needs a GoogleLoginService and GoogleServicesFramework running. While the μg version of GLS is already in a working version on git, the GSF is not. There were copyright issues with the μg GSF so  I needed to fix them up, but I never found time to do so. Moreover this version of Phonesky is not ready for everyday usage yet (actually nearly all frontend code is missing).
The old version is here: https://github.com/mar-v-in/BlankStore (i strongly suggest taking a look at cernekee's fork instead, he eased up the build a lot: https://github.com/cernekee/BlankStore)

If you're interested in future development, feel free to contact me via mail

**Comment by mid-kid** [comment]
No need to be sorry for not having a convenient build system. I never paid you to have one. 
I know this is not intended for daily use, I just wanted to see if it built.

As for the old BlankStore, your precompiled version works, if not for not having permission to install packages.
I have tried building cernekee's version, but it just gives me a Bad Request error (in logcat).
I'm going to try and build your version from source.

EDIT: Welp, looks like I may have entered the wrong device ID, because when I rebuilt cernekee's version, it worked.
